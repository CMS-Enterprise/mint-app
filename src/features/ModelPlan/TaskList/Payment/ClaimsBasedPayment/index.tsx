import React, { useRef } from 'react';
import { useTranslation } from 'react-i18next';
import { useHistory, useParams } from 'react-router-dom';
import {
  Button,
  Fieldset,
  Grid,
  GridContainer,
  Icon,
  Label
} from '@trussworks/react-uswds';
import { NotFoundPartial } from 'features/NotFound';
import { Field, Form, Formik, FormikProps } from 'formik';
import {
  ClaimsBasedPayType,
  GetClaimsBasedPaymentQuery,
  PayType,
  TypedUpdatePaymentsDocument,
  useGetClaimsBasedPaymentQuery
} from 'gql/generated/graphql';

import AddNote from 'components/AddNote';
import AskAQuestion from 'components/AskAQuestion';
import BooleanRadio from 'components/BooleanRadioForm';
import Breadcrumbs, { BreadcrumbItemOptions } from 'components/Breadcrumbs';
import ConfirmLeave from 'components/ConfirmLeave';
import FieldGroup from 'components/FieldGroup';
import MTOWarning from 'components/MTOWarning';
import MultiSelect from 'components/MultiSelect';
import MutationErrorModal from 'components/MutationErrorModal';
import PageHeading from 'components/PageHeading';
import PageNumber from 'components/PageNumber';
import TextAreaField from 'components/TextAreaField';
import TextField from 'components/TextField';
import useHandleMutation from 'hooks/useHandleMutation';
import usePlanTranslation from 'hooks/usePlanTranslation';
import useScrollElement from 'hooks/useScrollElement';
import { composeMultiSelectOptions } from 'utils/modelPlan';

import { renderCurrentPage, renderTotalPages } from '..';

type ClaimsBasedPaymentFormType =
  GetClaimsBasedPaymentQuery['modelPlan']['payments'];

const ClaimsBasedPayment = () => {
  const { t: paymentsT } = useTranslation('payments');

  const { t: paymentsMiscT } = useTranslation('paymentsMisc');

  const { t: miscellaneousT } = useTranslation('miscellaneous');

  const {
    payClaims: payClaimsConfig,
    shouldAnyProvidersExcludedFFSSystems:
      shouldAnyProvidersExcludedFFSSystemsConfig,
    changesMedicarePhysicianFeeSchedule:
      changesMedicarePhysicianFeeScheduleConfig,
    affectsMedicareSecondaryPayerClaims:
      affectsMedicareSecondaryPayerClaimsConfig
  } = usePlanTranslation('payments');

  const { modelID = '' } = useParams<{ modelID: string }>();

  const formikRef = useRef<FormikProps<ClaimsBasedPaymentFormType>>(null);
  const history = useHistory();

  const { data, loading, error } = useGetClaimsBasedPaymentQuery({
    variables: {
      id: modelID
    }
  });

  // If redirected from Operational Solutions, scrolls to the relevant question
  useScrollElement(!loading);

  const {
    id,
    payType,
    payClaims,
    payClaimsOther,
    payClaimsNote,
    shouldAnyProvidersExcludedFFSSystems,
    shouldAnyProviderExcludedFFSSystemsNote,
    changesMedicarePhysicianFeeSchedule,
    changesMedicarePhysicianFeeScheduleNote,
    affectsMedicareSecondaryPayerClaims,
    affectsMedicareSecondaryPayerClaimsHow,
    affectsMedicareSecondaryPayerClaimsNote,
    payModelDifferentiation
  } = (data?.modelPlan?.payments || {}) as ClaimsBasedPaymentFormType;

  const modelName = data?.modelPlan?.modelName || '';

  const { mutationError } = useHandleMutation(TypedUpdatePaymentsDocument, {
    id,
    formikRef
  });

  const initialValues: ClaimsBasedPaymentFormType = {
    __typename: 'PlanPayments',
    id: id ?? '',
    payType: payType ?? [],
    payClaims: payClaims ?? [],
    payClaimsOther: payClaimsOther ?? '',
    payClaimsNote: payClaimsNote ?? '',
    shouldAnyProvidersExcludedFFSSystems:
      shouldAnyProvidersExcludedFFSSystems ?? null,
    shouldAnyProviderExcludedFFSSystemsNote:
      shouldAnyProviderExcludedFFSSystemsNote ?? '',
    changesMedicarePhysicianFeeSchedule:
      changesMedicarePhysicianFeeSchedule ?? null,
    changesMedicarePhysicianFeeScheduleNote:
      changesMedicarePhysicianFeeScheduleNote ?? '',
    affectsMedicareSecondaryPayerClaims:
      affectsMedicareSecondaryPayerClaims ?? null,
    affectsMedicareSecondaryPayerClaimsHow:
      affectsMedicareSecondaryPayerClaimsHow ?? '',
    affectsMedicareSecondaryPayerClaimsNote:
      affectsMedicareSecondaryPayerClaimsNote ?? '',
    payModelDifferentiation: payModelDifferentiation ?? ''
  };

  if ((!loading && error) || (!loading && !data?.modelPlan)) {
    return <NotFoundPartial />;
  }

  return (
    <>
      <MutationErrorModal
        isOpen={mutationError.isModalOpen}
        closeModal={() => mutationError.setIsModalOpen(false)}
        url={mutationError.destinationURL}
      />

      <Breadcrumbs
        items={[
          BreadcrumbItemOptions.HOME,
          BreadcrumbItemOptions.COLLABORATION_AREA,
          BreadcrumbItemOptions.TASK_LIST,
          BreadcrumbItemOptions.PAYMENTS
        ]}
      />

      <PageHeading className="margin-top-4 margin-bottom-2">
        {paymentsMiscT('heading')}
      </PageHeading>

      <p
        className="margin-top-0 margin-bottom-1 font-body-lg"
        data-testid="model-plan-name"
      >
        {miscellaneousT('for')} {modelName}
      </p>

      <p className="margin-bottom-2 font-body-md line-height-sans-4">
        {miscellaneousT('helpText')}
      </p>

      <AskAQuestion modelID={modelID} />

      <Formik
        initialValues={initialValues}
        onSubmit={() => {
          history.push(
            `/models/${modelID}/collaboration-area/task-list/payment/anticipating-dependencies`
          );
        }}
        enableReinitialize
        innerRef={formikRef}
      >
        {(formikProps: FormikProps<ClaimsBasedPaymentFormType>) => {
          const { handleSubmit, setFieldValue, setErrors, values } =
            formikProps;

          return (
            <>
              <ConfirmLeave />

              <GridContainer className="padding-left-0 padding-right-0">
                <Grid row gap>
                  <Grid desktop={{ col: 6 }}>
                    <Form
                      className="margin-top-6"
                      data-testid="payment-claims-based-payment-form"
                      onSubmit={e => {
                        handleSubmit(e);
                      }}
                    >
                      <Fieldset disabled={!!error || loading}>
                        <PageHeading
                          headingLevel="h3"
                          className="margin-bottom-3"
                        >
                          {paymentsMiscT('claimSpecificQuestions')}
                        </PageHeading>

                        <FieldGroup className="margin-top-4">
                          <Label
                            htmlFor="payment-pay-claims"
                            id="label-payment-pay-claims"
                          >
                            {paymentsT('payClaims.label')}
                          </Label>

                          <p className="text-base margin-bottom-1 margin-top-05">
                            {paymentsT('payClaims.sublabel')}
                          </p>

                          <Field
                            as={MultiSelect}
                            id="payment-pay-claims"
                            name="payClaims"
                            ariaLabel="label-payment-pay-claims"
                            options={composeMultiSelectOptions(
                              payClaimsConfig.options
                            )}
                            selectedLabel={paymentsT(
                              'payClaims.multiSelectLabel'
                            )}
                            onChange={(value: string[] | []) => {
                              setFieldValue('payClaims', value);
                            }}
                            initialValues={initialValues.payClaims}
                          />

                          {(values?.payClaims || []).includes(
                            ClaimsBasedPayType.OTHER
                          ) && (
                            <FieldGroup>
                              <Label
                                htmlFor="payClaimsOther"
                                className="text-normal"
                              >
                                {paymentsT('payClaimsOther.label')}
                              </Label>

                              <Field
                                as={TextField}
                                id="payment-pay-claims-other"
                                data-testid="payment-pay-claims-other"
                                name="payClaimsOther"
                              />
                            </FieldGroup>
                          )}

                          <AddNote id="pay-claims-note" field="payClaimsNote" />
                        </FieldGroup>

                        <FieldGroup
                          className="margin-top-4"
                          scrollElement="shouldAnyProvidersExcludedFFSSystems"
                        >
                          <Label
                            htmlFor="shouldAnyProvidersExcludedFFSSystems"
                            className="maxw-none"
                          >
                            {paymentsT(
                              'shouldAnyProvidersExcludedFFSSystems.label'
                            )}
                          </Label>

                          <MTOWarning
                            id="payment-provider-exclusion-ffs-system-warning"
                            className="margin-top-neg-5"
                          />

                          <BooleanRadio
                            field="shouldAnyProvidersExcludedFFSSystems"
                            id="payment-provider-exclusion-ffs-system"
                            value={values.shouldAnyProvidersExcludedFFSSystems}
                            setFieldValue={setFieldValue}
                            options={
                              shouldAnyProvidersExcludedFFSSystemsConfig.options
                            }
                          />

                          <AddNote
                            id="payment-provider-exclusion-ffs-system-note"
                            field="shouldAnyProviderExcludedFFSSystemsNote"
                          />
                        </FieldGroup>

                        <FieldGroup className="margin-top-4">
                          <Label
                            htmlFor="changesMedicarePhysicianFeeSchedule"
                            className="maxw-none"
                          >
                            {paymentsT(
                              'changesMedicarePhysicianFeeSchedule.label'
                            )}
                          </Label>

                          <p className="text-base margin-y-1">
                            {paymentsT(
                              'changesMedicarePhysicianFeeSchedule.sublabel'
                            )}
                          </p>

                          <BooleanRadio
                            field="changesMedicarePhysicianFeeSchedule"
                            id="payment-change-medicare-phyisican-fee-schedule"
                            value={values.changesMedicarePhysicianFeeSchedule}
                            setFieldValue={setFieldValue}
                            options={
                              changesMedicarePhysicianFeeScheduleConfig.options
                            }
                          />

                          <AddNote
                            id="payment-change-medicare-phyisican-fee-schedule-note"
                            field="changesMedicarePhysicianFeeScheduleNote"
                          />
                        </FieldGroup>

                        <FieldGroup className="margin-top-4">
                          <Label
                            htmlFor="affectsMedicareSecondaryPayerClaims"
                            className="maxw-none"
                          >
                            {paymentsT(
                              'affectsMedicareSecondaryPayerClaims.label'
                            )}
                          </Label>

                          <BooleanRadio
                            field="affectsMedicareSecondaryPayerClaims"
                            id="payment-affects-medicare-secondary-payer-claims"
                            value={values.affectsMedicareSecondaryPayerClaims}
                            setFieldValue={setFieldValue}
                            options={
                              affectsMedicareSecondaryPayerClaimsConfig.options
                            }
                            childName="affectsMedicareSecondaryPayerClaimsHow"
                          >
                            {values.affectsMedicareSecondaryPayerClaims ? (
                              <FieldGroup className="margin-left-4 margin-y-1">
                                <Label
                                  htmlFor="payment-affects-medicare-secondary-payer-claims-how"
                                  className="text-normal"
                                >
                                  {paymentsT(
                                    'affectsMedicareSecondaryPayerClaimsHow.label'
                                  )}
                                </Label>

                                <Field
                                  as={TextAreaField}
                                  className="height-15"
                                  id="payment-affects-medicare-secondary-payer-claims-how"
                                  data-testid="payment-affects-medicare-secondary-payer-claims-how"
                                  name="affectsMedicareSecondaryPayerClaimsHow"
                                />
                              </FieldGroup>
                            ) : (
                              <></>
                            )}
                          </BooleanRadio>

                          <AddNote
                            id="payment-affects-medicare-secondary-payer-claims-note"
                            field="affectsMedicareSecondaryPayerClaimsNote"
                          />
                        </FieldGroup>

                        <FieldGroup className="margin-top-4">
                          <Label
                            htmlFor="payModelDifferentiation"
                            className="maxw-none"
                          >
                            {paymentsT('payModelDifferentiation.label')}
                          </Label>

                          <Field
                            as={TextAreaField}
                            className="height-15"
                            id="payment-affect-current-policy"
                            data-testid="payment-affect-current-policy"
                            name="payModelDifferentiation"
                          />
                        </FieldGroup>

                        <div className="margin-top-6 margin-bottom-3">
                          <Button
                            type="button"
                            className="usa-button usa-button--outline margin-bottom-1"
                            onClick={() => {
                              history.push(
                                `/models/${modelID}/collaboration-area/task-list/payment`
                              );
                            }}
                          >
                            {miscellaneousT('back')}
                          </Button>

                          <Button type="submit" onClick={() => setErrors({})}>
                            {miscellaneousT('next')}
                          </Button>
                        </div>

                        <Button
                          type="button"
                          className="usa-button usa-button--unstyled"
                          onClick={() =>
                            history.push(
                              `/models/${modelID}/collaboration-area/task-list`
                            )
                          }
                        >
                          <Icon.ArrowBack
                            className="margin-right-1"
                            aria-hidden
                            aria-label="back"
                          />

                          {miscellaneousT('saveAndReturn')}
                        </Button>
                      </Fieldset>
                    </Form>
                  </Grid>
                </Grid>
              </GridContainer>
            </>
          );
        }}
      </Formik>

      {data && (
        <PageNumber
          currentPage={renderCurrentPage(
            2,
            payType.includes(PayType.CLAIMS_BASED_PAYMENTS),
            payType.includes(PayType.NON_CLAIMS_BASED_PAYMENTS),
            payClaims.includes(
              ClaimsBasedPayType.REDUCTIONS_TO_BENEFICIARY_COST_SHARING
            )
          )}
          totalPages={renderTotalPages(
            payType.includes(PayType.CLAIMS_BASED_PAYMENTS),
            payType.includes(PayType.NON_CLAIMS_BASED_PAYMENTS),
            payClaims.includes(
              ClaimsBasedPayType.REDUCTIONS_TO_BENEFICIARY_COST_SHARING
            )
          )}
          className="margin-y-6"
        />
      )}
    </>
  );
};

export default ClaimsBasedPayment;
