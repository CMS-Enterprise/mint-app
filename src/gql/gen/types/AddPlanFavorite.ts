/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL mutation operation: AddPlanFavorite
// ====================================================

export interface AddPlanFavorite_addPlanFavorite {
  __typename: "PlanFavorite";
  modelPlanID: UUID;
  userID: UUID;
}

export interface AddPlanFavorite {
  addPlanFavorite: AddPlanFavorite_addPlanFavorite;
}

export interface AddPlanFavoriteVariables {
  modelPlanID: UUID;
}
