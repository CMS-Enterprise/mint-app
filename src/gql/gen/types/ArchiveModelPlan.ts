/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL mutation operation: ArchiveModelPlan
// ====================================================

export interface ArchiveModelPlan_updateModelPlan {
  __typename: "ModelPlan";
  id: UUID;
}

export interface ArchiveModelPlan {
  updateModelPlan: ArchiveModelPlan_updateModelPlan;
}

export interface ArchiveModelPlanVariables {
  id: UUID;
  archived: boolean;
}
