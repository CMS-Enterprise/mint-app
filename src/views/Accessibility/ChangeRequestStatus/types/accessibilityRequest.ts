/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { AccessibilityRequestStatus } from "./../../../../types/graphql-global-types";

// ====================================================
// GraphQL query operation: accessibilityRequest
// ====================================================

export interface accessibilityRequest_accessibilityRequest_statusRecord {
  __typename: "AccessibilityRequestStatusRecord";
  status: AccessibilityRequestStatus;
}

export interface accessibilityRequest_accessibilityRequest {
  __typename: "AccessibilityRequest";
  id: UUID;
  name: string;
  statusRecord: accessibilityRequest_accessibilityRequest_statusRecord;
}

export interface accessibilityRequest {
  accessibilityRequest: accessibilityRequest_accessibilityRequest | null;
}

export interface accessibilityRequestVariables {
  id: UUID;
}
