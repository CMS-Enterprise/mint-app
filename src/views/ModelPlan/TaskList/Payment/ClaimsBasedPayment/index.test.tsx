import React from 'react';
import { MemoryRouter, Route } from 'react-router-dom';
import { MockedProvider } from '@apollo/client/testing';
import { render, screen, waitFor } from '@testing-library/react';
import {
  GetClaimsBasedPaymentDocument,
  GetClaimsBasedPaymentQuery
} from 'gql/gen/graphql';

import { ClaimsBasedPayType, PayType } from 'types/graphql-global-types';

import ClaimsBasedPayment from './index';

type GetClaimsBasedPaymentType = GetClaimsBasedPaymentQuery['modelPlan']['payments'];

const mockData: GetClaimsBasedPaymentType = {
  __typename: 'PlanPayments',
  id: '123',
  payType: [PayType.CLAIMS_BASED_PAYMENTS],
  payClaims: [ClaimsBasedPayType.OTHER],
  payClaimsNote: '',
  payClaimsOther: 'pay claims other',
  shouldAnyProvidersExcludedFFSSystems: null,
  shouldAnyProviderExcludedFFSSystemsNote: '',
  changesMedicarePhysicianFeeSchedule: null,
  changesMedicarePhysicianFeeScheduleNote: '',
  affectsMedicareSecondaryPayerClaims: null,
  affectsMedicareSecondaryPayerClaimsHow: '',
  affectsMedicareSecondaryPayerClaimsNote: '',
  payModelDifferentiation: ''
};

const paymentsMock = [
  {
    request: {
      query: GetClaimsBasedPaymentDocument,
      variables: { id: 'ce3405a0-3399-4e3a-88d7-3cfc613d2905' }
    },
    result: {
      data: {
        modelPlan: {
          id: 'ce3405a0-3399-4e3a-88d7-3cfc613d2905',
          modelName: 'My excellent plan that I just initiated',
          payments: mockData,
          operationalNeeds: [
            {
              __typename: 'OperationalNeed',
              id: '424213',
              modifiedDts: null
            }
          ]
        }
      }
    }
  }
];

describe('Model Plan -- Claims Based Payment', () => {
  it('renders without errors', async () => {
    render(
      <MemoryRouter
        initialEntries={[
          '/models/ce3405a0-3399-4e3a-88d7-3cfc613d2905/task-list/payment/claims-based-payment'
        ]}
      >
        <MockedProvider mocks={paymentsMock} addTypename={false}>
          <Route path="/models/:modelID/task-list/payment/claims-based-payment">
            <ClaimsBasedPayment />
          </Route>
        </MockedProvider>
      </MemoryRouter>
    );

    await waitFor(() => {
      expect(
        screen.getByTestId('payment-claims-based-payment-form')
      ).toBeInTheDocument();
    });

    await waitFor(() => {
      expect(screen.getByTestId('payment-pay-claims-other')).toHaveValue(
        'pay claims other'
      );
    });
  });

  it('matches snapshot', async () => {
    const { asFragment } = render(
      <MemoryRouter
        initialEntries={[
          '/models/ce3405a0-3399-4e3a-88d7-3cfc613d2905/task-list/payment/claims-based-payment'
        ]}
      >
        <MockedProvider mocks={paymentsMock} addTypename={false}>
          <Route path="/models/:modelID/task-list/payment/claims-based-payment">
            <ClaimsBasedPayment />
          </Route>
        </MockedProvider>
      </MemoryRouter>
    );

    await waitFor(() => {
      expect(
        screen.getByTestId('payment-claims-based-payment-form')
      ).toBeInTheDocument();
    });
    expect(asFragment()).toMatchSnapshot();
  });
});
