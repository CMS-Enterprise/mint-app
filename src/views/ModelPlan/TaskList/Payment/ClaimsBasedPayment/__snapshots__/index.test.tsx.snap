// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Model Plan -- Claims Based Payment matches snapshot 1`] = `
<DocumentFragment>
  <nav
    aria-label="Breadcrumbs"
    class="usa-breadcrumb usa-breadcrumb--wrap"
  >
    <ol
      class="usa-breadcrumb__list"
    >
      <li
        class="usa-breadcrumb__list-item"
      >
        <a
          class="usa-breadcrumb__link"
          href="/"
        >
          <span>
            Home
          </span>
        </a>
      </li>
      <li
        class="usa-breadcrumb__list-item"
      >
        <a
          class="usa-breadcrumb__link"
          href="/models/ce3405a0-3399-4e3a-88d7-3cfc613d2905/task-list/"
        >
          <span>
            Model Plan task list
          </span>
        </a>
      </li>
      <li
        aria-current="page"
        class="usa-breadcrumb__list-item usa-current"
      >
        Payment
      </li>
    </ol>
  </nav>
  <h1
    aria-live="polite"
    class="mint-h1 margin-top-4 margin-bottom-2"
    tabindex="-1"
  >
    Payment
  </h1>
  <p
    class="margin-top-0 margin-bottom-1 font-body-lg"
    data-testid="model-plan-name"
  >
    for 
  </p>
  <p
    class="margin-bottom-2 font-body-md line-height-sans-4"
  >
    If there's a question or field that is not applicable to your model or you don't currently know the answer, you may leave it blank. If you need help, ask a question using the link below.
  </p>
  <div
    class="padding-2 bg-primary-lighter"
  >
    <div
      class="display-flex"
      data-testid="ask-a-question"
    >
      <svg
        class="usa-icon text-primary margin-right-1"
        focusable="false"
        height="1em"
        role="img"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M20 2H4c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-7 9h-2V5h2v6zm0 4h-2v-2h2v2z"
        />
      </svg>
      <button
        class="usa-button usa-button--unstyled"
        data-testid="ask-a-question-button"
        type="button"
      >
        Ask a question
      </button>
    </div>
  </div>
  <div
    class="grid-container padding-left-0 padding-right-0"
    data-testid="gridContainer"
  >
    <div
      class="grid-row grid-gap"
      data-testid="grid"
    >
      <div
        class="desktop:grid-col-6"
        data-testid="grid"
      >
        <form
          action="#"
          class="margin-top-6"
          data-testid="payment-claims-based-payment-form"
        >
          <h3
            aria-live="polite"
            class="mint-h1 margin-bottom-3"
            tabindex="-1"
          >
            Claims-based payment questions
          </h3>
          <div
            class="usa-form-group margin-top-4"
            data-scroll="payClaims"
          >
            <label
              class="usa-label"
              data-testid="label"
              for="payment-pay-claims"
              id="label-payment-pay-claims"
            >
              Select which claims-based payments will you pay.
            </label>
            <p
              class="text-base margin-bottom-1 margin-top-05"
            >
              If you select reductions to beneficiary cost-sharing or other, there will be additional questions to answer.
            </p>
            <div>
              <div
                class="easi-multiselect usa-combo-box margin-top-1 css-b62m3t-container"
                id="payment-pay-claims"
              >
                <span
                  class="css-1f43avz-a11yText-A11yText"
                  id="react-select-3-live-region"
                />
                <span
                  aria-atomic="false"
                  aria-live="polite"
                  aria-relevant="additions text"
                  class="css-1f43avz-a11yText-A11yText"
                />
                <div
                  class=" css-jvusj8-control"
                >
                  <div
                    class=" css-319lph-ValueContainer"
                  >
                    <div
                      class=" css-14el2xx-placeholder"
                      id="react-select-3-placeholder"
                    >
                      0 selected
                    </div>
                    <div
                      class=" css-6j8wv5-Input"
                      data-value=""
                    >
                      <input
                        aria-autocomplete="list"
                        aria-describedby="react-select-3-placeholder"
                        aria-expanded="false"
                        aria-haspopup="true"
                        aria-labelledby="label-payment-pay-claims"
                        autocapitalize="none"
                        autocomplete="off"
                        autocorrect="off"
                        class=""
                        id="react-select-3-input"
                        role="combobox"
                        spellcheck="false"
                        style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                        tabindex="0"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class=" css-1hb7zxy-IndicatorsContainer"
                  >
                    <span
                      class=" css-1vtyy7d-indicatorSeparator"
                    />
                    <div
                      aria-hidden="true"
                      class=" css-1232v1p-indicatorContainer"
                    >
                      <svg
                        aria-hidden="true"
                        class="css-tj5bde-Svg"
                        focusable="false"
                        height="20"
                        viewBox="0 0 20 20"
                        width="20"
                      >
                        <path
                          d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
                <div>
                  <input
                    name="payClaims"
                    type="hidden"
                  />
                </div>
              </div>
            </div>
            <div
              class="margin-top-4 margin-bottom-8"
            >
              <button
                class="usa-button usa-button usa-button--unstyled"
                data-testid="add-note-toggle"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="usa-icon margin-right-1"
                  focusable="false"
                  height="1em"
                  role="img"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                  />
                </svg>
                Add an additional note
              </button>
            </div>
          </div>
          <div
            class="usa-form-group margin-top-4"
            data-scroll="shouldAnyProvidersExcludedFFSSystems"
          >
            <label
              class="usa-label maxw-none"
              data-testid="label"
              for="shouldAnyProvidersExcludedFFSSystems"
            >
              Should any model providers be excluded from existing Fee-for-Service payment systems?
            </label>
            <fieldset
              class="usa-fieldset"
              data-testid="fieldset"
            >
              <div
                class="usa-radio"
                data-testid="radio"
              >
                <input
                  class="usa-radio__input"
                  data-testid="payment-provider-exclusion-ffs-system-true"
                  id="payment-provider-exclusion-ffs-system-true"
                  name="shouldAnyProvidersExcludedFFSSystems"
                  type="radio"
                  value="YES"
                />
                <label
                  class="usa-radio__label"
                  for="payment-provider-exclusion-ffs-system-true"
                >
                  Yes
                </label>
              </div>
              <div
                class="usa-radio"
                data-testid="radio"
              >
                <input
                  class="usa-radio__input"
                  data-testid="payment-provider-exclusion-ffs-system-false"
                  id="payment-provider-exclusion-ffs-system-false"
                  name="shouldAnyProvidersExcludedFFSSystems"
                  type="radio"
                  value="NO"
                />
                <label
                  class="usa-radio__label"
                  for="payment-provider-exclusion-ffs-system-false"
                >
                  No
                </label>
              </div>
            </fieldset>
            <div
              class="margin-top-4 margin-bottom-8"
            >
              <button
                class="usa-button usa-button usa-button--unstyled"
                data-testid="add-note-toggle"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="usa-icon margin-right-1"
                  focusable="false"
                  height="1em"
                  role="img"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                  />
                </svg>
                Add an additional note
              </button>
            </div>
          </div>
          <div
            class="usa-form-group margin-top-4"
            data-scroll="changesMedicarePhysicianFeeSchedule"
          >
            <label
              class="usa-label maxw-none"
              data-testid="label"
              for="changesMedicarePhysicianFeeSchedule"
            >
              Will the model change the Medicare Physician Fee Schedule?
            </label>
            <p
              class="text-base margin-y-1"
            >
              If so, it might impact Critical Access Hospitals paid under Method II.
            </p>
            <fieldset
              class="usa-fieldset"
              data-testid="fieldset"
            >
              <div
                class="usa-radio"
                data-testid="radio"
              >
                <input
                  class="usa-radio__input"
                  data-testid="payment-change-medicare-phyisican-fee-schedule-true"
                  id="payment-change-medicare-phyisican-fee-schedule-true"
                  name="changesMedicarePhysicianFeeSchedule"
                  type="radio"
                  value="YES"
                />
                <label
                  class="usa-radio__label"
                  for="payment-change-medicare-phyisican-fee-schedule-true"
                >
                  Yes
                </label>
              </div>
              <div
                class="usa-radio"
                data-testid="radio"
              >
                <input
                  class="usa-radio__input"
                  data-testid="payment-change-medicare-phyisican-fee-schedule-false"
                  id="payment-change-medicare-phyisican-fee-schedule-false"
                  name="changesMedicarePhysicianFeeSchedule"
                  type="radio"
                  value="NO"
                />
                <label
                  class="usa-radio__label"
                  for="payment-change-medicare-phyisican-fee-schedule-false"
                >
                  No
                </label>
              </div>
            </fieldset>
            <div
              class="margin-top-4 margin-bottom-8"
            >
              <button
                class="usa-button usa-button usa-button--unstyled"
                data-testid="add-note-toggle"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="usa-icon margin-right-1"
                  focusable="false"
                  height="1em"
                  role="img"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                  />
                </svg>
                Add an additional note
              </button>
            </div>
          </div>
          <div
            class="usa-form-group margin-top-4"
            data-scroll="affectsMedicareSecondaryPayerClaims"
          >
            <label
              class="usa-label maxw-none"
              data-testid="label"
              for="affectsMedicareSecondaryPayerClaims"
            >
              Does the model affect Medicare Secondary Payer claims?
            </label>
            <fieldset
              class="usa-fieldset"
              data-testid="fieldset"
            >
              <div
                class="usa-radio"
                data-testid="radio"
              >
                <input
                  class="usa-radio__input"
                  id="payment-affects-medicare-secondary-payer-claims-Yes"
                  name="affectsMedicareSecondaryPayerClaims"
                  type="radio"
                  value="YES"
                />
                <label
                  class="usa-radio__label"
                  for="payment-affects-medicare-secondary-payer-claims-Yes"
                >
                  Yes
                </label>
              </div>
              <div
                class="usa-radio"
                data-testid="radio"
              >
                <input
                  class="usa-radio__input"
                  id="payment-affects-medicare-secondary-payer-claims-No"
                  name="affectsMedicareSecondaryPayerClaims"
                  type="radio"
                  value="FALSE"
                />
                <label
                  class="usa-radio__label"
                  for="payment-affects-medicare-secondary-payer-claims-No"
                >
                  No
                </label>
              </div>
            </fieldset>
            <div
              class="margin-top-4 margin-bottom-8"
            >
              <button
                class="usa-button usa-button usa-button--unstyled"
                data-testid="add-note-toggle"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="usa-icon margin-right-1"
                  focusable="false"
                  height="1em"
                  role="img"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                  />
                </svg>
                Add an additional note
              </button>
            </div>
          </div>
          <div
            class="usa-form-group margin-top-4"
            data-scroll="payModelDifferentiation"
          >
            <label
              class="usa-label maxw-none"
              data-testid="label"
              for="payModelDifferentiation"
            >
              How does the model differ from current policy, especially in terms of payment rates and periodicity of payment?
            </label>
            <textarea
              aria-labelledby="label-payment-affect-current-policy hint-payment-affect-current-policy"
              class="height-15 usa-textarea"
              data-testid="payment-affect-current-policy"
              id="payment-affect-current-policy"
              name="payModelDifferentiation"
            />
          </div>
          <div
            class="margin-top-6 margin-bottom-3"
          >
            <button
              class="usa-button usa-button usa-button--outline margin-bottom-1"
              data-testid="button"
              type="button"
            >
              Back
            </button>
            <button
              class="usa-button"
              data-testid="button"
              type="submit"
            >
              Next
            </button>
          </div>
          <button
            class="usa-button usa-button usa-button--unstyled"
            data-testid="button"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="usa-icon margin-right-1"
              focusable="false"
              height="1em"
              role="img"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
              />
            </svg>
            Save and return to task list
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
